app/controllers/concerns/api_helper.rb

```diff
@@ -11,29 +11,38 @@ module ApiHelper
 
   CHECKSUM_LENGTH_SHA1 = 40
   CHECKSUM_LENGTH_SHA256 = 64
+  CHECKSUM_LENGTH_SHA512 = 128
 
   # Verify checksum
   def verify_checksum
+    secrets = Rails.configuration.x.loadbalancer_secrets
     raise ChecksumError unless params[:checksum].present? &&
                                (params[:checksum].length == CHECKSUM_LENGTH_SHA1 ||
-                                params[:checksum].length == CHECKSUM_LENGTH_SHA256)
+                                params[:checksum].length == CHECKSUM_LENGTH_SHA256 ||
+                                params[:checksum].length == CHECKSUM_LENGTH_SHA512) &&
+                               secrets.any?
 
     # Camel case (ex) get_meetings to getMeetings to match BBB server
     check_string = action_name.camelcase(:lower)
     check_string += request.query_string.gsub(
       /&checksum=#{params[:checksum]}|checksum=#{params[:checksum]}&|checksum=#{params[:checksum]}/, ''
     )
 
-    return if Rails.configuration.x.loadbalancer_secrets.any? do |secret|
-      checksum_sha1 = Digest::SHA1.hexdigest(check_string + secret)
-      checksum_sha256 = Digest::SHA256.hexdigest(check_string + secret)
-      ActiveSupport::SecurityUtils.secure_compare(checksum_sha1, params[:checksum]) ||
-      ActiveSupport::SecurityUtils.secure_compare(checksum_sha256, params[:checksum])
+    checksum_algorithms = Rails.configuration.x.loadbalancer_checksum_algorithms
+    secrets.product(checksum_algorithms).any? do |secret, checksum_algorithm|
+      return true if ActiveSupport::SecurityUtils.secure_compare(get_checksum(check_string + secret, checksum_algorithm),
+      params[:checksum])
     end
 
     raise ChecksumError
   end
 
+  def get_checksum(check_string, checksum_algorithm)
+    return Digest::SHA512.hexdigest(check_string) if checksum_algorithm == 'SHA512'
```

Comment:
This should be reformatted as a `case` statement. Consider setting a default value for the `checksum_algorithm` option.

The method should raise an exception if an unsupported value of `checksum_algorithm` is passed.

Reply:
yeah, it would work that way too